python/lsst/daf/butler/instrument.py

```diff
@@ -0,0 +1,70 @@
+# This file is part of daf_butler.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+
+__all__ = ("Instrument", )
+
+
+class Instrument:
+    """A template method class that can register itself with a
+    `Registry.
+
+    This class should be subclassed by various implementations.
+
+    Attributes
+    ----------
+    camera : `str`
+        Name of the camera.
+    physicalFilters : `list`
+        List of PhysicalFilter entries (each entry being a dict).
+    sensors : `list`
+        List of Sensor entries (each entry being a dict).
+    """
+    @property
+    def camera(self):
+        raise NotImplementedError('Must be specified by derived class')
+
+    physicalFilters = []
+    sensors = []
```

Comment:
It doesn't really make sense to me that `camera` is an instance property while the others are class attributes.

I think I'd probably just make `camera` a class attribute that's set to `None`, and document in the class that all that matters is that subclass instances have these attributes; that will let subclasses that need to override them at the instance level while allowing those that don't to override them at the class level.